tests/sentry/workflow_engine/processors/test_delayed_workflow.py

```diff
@@ -639,3 +654,129 @@ def test_multiple_dcgs_per_group(self):
+        )
+        self.create_data_condition_group_action(
+            condition_group=self.workflow1_dcgs[1], action=action1
+        )
+
+        # action2 = self.create_action(
+        #     type=Action.Type.SLACK,
+        #     integration_id="1234567890",
+        #     target_identifier="channel789",
+        #     target_display="#general",
+        #     data={"tags": "environment,user", "notes": "Important alert"},
+        # )
+        action2 = self.create_action(
+            type=Action.Type.DISCORD,
+            integration_id="1234567890",
+            target_identifier="channel456",
+            data={"tags": "environment,user,my_tag"},
+        )
+        self.create_data_condition_group_action(
+            condition_group=self.workflow2_dcgs[1], action=action2
+        )
+
+        self.groups_to_dcgs = {
+            self.group1.id: set(self.workflow1_dcgs),
+            self.group2.id: set(self.workflow2_dcgs),
+        }
+        self.dcg_group_to_event_data: DataConditionGroupEvent = {}
+        for i, dcg in enumerate(self.workflow_dcgs):
+            if i < 2:
+                group = self.group1
+                event = self.event1
+            else:
+                group = self.group2
+                event = self.event2
+
+            self.dcg_group_to_event_data[(dcg.id, group.id)] = {
+                "event_id": event.event_id,
+                "occurrence_id": None,
+            }
+
+        self.group_to_groupevent = {
+            self.group1: self.event1.for_group(self.group1),
+            self.group2: self.event2.for_group(self.group2),
+        }
+
+    def test_parse_dcg_group_event_data(self):
+        self._push_base_events()
+        buffer_data = fetch_group_to_event_data(self.project.id, Workflow)
+        dcg_group_to_event_data, event_ids, occurrence_ids = parse_dcg_group_event_data(
+            buffer_data, self.groups_to_dcgs
+        )
+
+        assert dcg_group_to_event_data == self.dcg_group_to_event_data
+        assert event_ids == {self.event1.event_id, self.event2.event_id}
+        assert occurrence_ids == set()
+
+    def test_bulk_fetch_events(self):
+        event_ids = [self.event1.event_id, self.event2.event_id]
+        events = bulk_fetch_events(event_ids, self.project.id)
+
+        for event in list(events.values()):
+            assert event.event_id in event_ids
+
+    def test_get_group_to_groupevent(self):
+        group_to_groupevent = get_group_to_groupevent(
+            self.dcg_group_to_event_data,
+            [self.group1.id, self.group2.id],
+            {self.event1.event_id, self.event2.event_id},
+            set(),
+            self.project.id,
+        )
+        assert group_to_groupevent == self.group_to_groupevent
+
+    @patch("sentry.workflow_engine.models.action.Action.trigger")
+    def test_fire_actions_for_groups__fire_actions(self, mock_trigger):
+        # only fires each action once, even though both the triggers and filters pass
```

Comment:
will we be able to assert that it only fired once later on?

Reply:
i don't think this is actually testing what i want, if a trigger with slow conditions passes we will always end up enqueuing the filter 🤔

Reply:
removing comment